### PR TITLE
create,edit,delete of compaign

### PR DESCRIPTION
**Pull request by ishrartx**

### CODE REVIEW

File: src/test/java/SalesforceModules/Compaignutil.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+import java.util.Random;
+
+public class Compaignutil {
+
+    //P[@class='slds-truncate'][text()='Community']
+
+
+
+
+    public static void clickComapignTab(String logStep) throws InterruptedException {
```
Comment (snaphray) [inline src/test/java/SalesforceModules/Compaignutil.java]: Please correct the spelling of Campaign

File: src/test/java/SalesforceModules/Compaignutil.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import pageobjects.SalesforceObjects.Comapign_object;
+import step_definitions.RunCukesTest;
+import utilities.HTMLReportUtil;
+import utilities.KeywordUtil;
+
+import java.util.Random;
+
+public class Compaignutil {
+
+    //P[@class='slds-truncate'][text()='Community']
```
Comment (snaphray) [inline src/test/java/SalesforceModules/Compaignutil.java]: Remove the commented line if not required.

File: src/test/java/SalesforceModules/Compaignutil.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+    public static void clickComapignTab(String logStep) throws InterruptedException {
+        KeywordUtil.delay(3000);
+        KeywordUtil.waitForVisible(Comapign_object.Comapign_Tab);
+        WebElement element = KeywordUtil.getDriver().findElement(Comapign_object.Comapign_Tab);
+        JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+    }
+
+    public static void entercompaignname(String compaignname, String logstep) {
```
Comment (snaphray) [inline src/test/java/SalesforceModules/Compaignutil.java]: Please follow proper naming convention for methods, change the method name to enterCampaignName.

Same applies to argumentsNames

File: src/test/java/SalesforceModules/Compaignutil.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+    public static void clickSaveButton(String logStep) throws InterruptedException {
+        KeywordUtil.delay(3000);
+        KeywordUtil.waitForVisible(Comapign_object.Save_Button);
+        WebElement element = KeywordUtil.getDriver().findElement(Comapign_object.Save_Button);
+        JavascriptExecutor executor = (JavascriptExecutor) KeywordUtil.getDriver();
+        executor.executeScript("arguments[0].click();", element);
+
+
+//        JavascriptExecutor executor = (JavascriptExecutor)KeywordUtil.getDriver();
```
Comment (snaphray) [inline src/test/java/SalesforceModules/Compaignutil.java]: Remove the commented line if not required

File: src/test/java/SalesforceModules/Compaignutil.java [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+//        executor.executeScript("arguments[0].click();", KeywordUtil.getDriver().findElement(AccountObject.saveButton));
+    }
+
+    public static String generateRandomString(int length) {
```
Comment (snaphray) [inline src/test/java/SalesforceModules/Compaignutil.java]: This method should be in KeywordUtil

File: src/test/java/SalesforceModules/ContactUtil.java [hunk 6]
```diff
@@ -124,6 +124,5 @@ public static void createContact(String salutation, String lastname, String emai
         enterEmail(email,email+" email entered");
         enterPhoneContact(phoneNumber,phoneNumber+" phone number entered");
         clickSaveButton("save button clicked");
```
Comment (snaphray) [inline src/test/java/SalesforceModules/ContactUtil.java:126]: why you have removed this line? is it not required?

File: src/test/java/pageobjects/SalesforceObjects/Comapign_object.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+public class Comapign_object {
+
+    public static By Comapign_Tab = By.xpath("//span[@class='slds-truncate'][text()='Campaigns']");
+
+    public static  By Comapign_Name= By.xpath("//div[@class=\"actionBody\"]//span[text()='Campaign Name']//parent::label//following-sibling::input");
+
+    //div[@class="actionBody"]//span[text()='Campaign Name']//parent::label//following-sibling::input
```
Comment (snaphray) [inline src/test/java/pageobjects/SalesforceObjects/Comapign_object.java]: Remove the commented line if not required

File: src/test/java/pageobjects/SalesforceObjects/Comapign_object.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+    public static By Save_Button= By.xpath("//button[@title=\"Save\"]");
```
Comment (snaphray) [inline src/test/java/pageobjects/SalesforceObjects/Comapign_object.java]: follow proper NamingConvention for object names and follow it all along

File: src/test/java/pageobjects/SalesforceObjects/Comapign_object.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+    public static By Show_Actions=By.xpath("//table[@role=\"grid\"]//tr//th[@scope=\"row\" ]//a[text()='DEF']//ancestor::th//following-sibling::td[8]//a//span[text()='Show Actions']");
+
+    public static By Show_Actions_Edit_Button=By.xpath("//a[@title='Edit']");
+
+    public static By Show_Actions_Delete_button=By.xpath("(//a[@title='Delete'])[2]");
+
+    public static By compaign_Delete_button=By.xpath("//span[text()='Delete']");
+
+    public static By select_option(String option) {
```
Comment (snaphray) [inline src/test/java/pageobjects/SalesforceObjects/Comapign_object.java]: Methods should not be present in the object file, please move them to appropriate location

File: src/test/resources/features/SalesMarketingCRM.feature [hunk 10]
```diff
@@ -0,0 +1,28 @@
+@Web
+Feature: Salesforce Marketing CRM
+
+
+  @MarketitingCRM_01
```
Comment (snaphray) [inline src/test/resources/features/SalesMarketingCRM.feature]: Check the spelling of tag and correct it

**Review by snaphray — CHANGES_REQUESTED** [review]
Please look at the comments and resolve them. Go through the coding guidelines document once again and follow it properly.

File: src/test/java/pageobjects/SalesforceObjects/Comapign_object.java (hunk 9 above)
Comment (ishrartx) [reply src/test/java/pageobjects/SalesforceObjects/Comapign_object.java]: This is a locator method where we are selecting the Values by passing in the cucumber file
like in the cucumber file we are passing the And user selects the "Marketing CRM Classic"
So if we want to change the name we have to just change on the cucumber file

**Review by snaphray — APPROVED** [review]
code looks fine

**Pull request merged**